### PR TITLE
Clone

### PR DESCRIPTION
**Pull request by hector0513**

### CODE REVIEW

File: src/event-listeners/index.js [hunk 1]
```diff
@@ -48,6 +53,7 @@ export default function setupEventListeners(store, canvas) {
 
   function bodyKeydown(event) {
     let state = store.getState()
+    console.log(state.ui.goalClone)
```
Comment (Connoropolous) [inline src/event-listeners/index.js]:
```suggestion
```

File: src/event-listeners/index.js [hunk 2]
```diff
@@ -93,6 +149,9 @@ export default function setupEventListeners(store, canvas) {
       case 'ShiftRight':
         store.dispatch(unsetShiftKeyDown())
         break
+      case 'ControlLeft':
```
Comment (Connoropolous) [inline src/event-listeners/index.js:125]:
```suggestion
      case 'ControlLeft':
      case 'ControlRight':
```

we should handle both keyboard keys, just like with ShiftLeft and ShiftRight, just above this ^^

File: src/event-listeners/index.js [hunk 3]
```diff
@@ -77,6 +83,56 @@ export default function setupEventListeners(store, canvas) {
           // prevent the browser from navigating back to the last page
           event.preventDefault()
         }
+      case 'ControlLeft':
```
Comment (Connoropolous) [inline src/event-listeners/index.js:85]:
```suggestion
      case 'ControlLeft':
      case 'ControlRight':
```

we should handle both keyboard keys, just like with ShiftLeft and ShiftRight, just above this ^^

File: src/holochainConfig.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
 export const DEVELOPMENT_INSTANCE_NAME = 'acorn'
 export const ZOME_NAME = 'holo_acorn'
 export const DEFAULT_HOLOCHAIN_PORT = '8888'
-export const DEFAULT_HOLOCHAIN_HOST = '127.0.0.1'
+export const DEFAULT_HOLOCHAIN_HOST = '192.168.1.61'
```
Comment (Connoropolous) [inline src/holochainConfig.js]:
```suggestion
export const DEFAULT_HOLOCHAIN_HOST = '127.0.0.1'
```

File: src/event-listeners/index.js [hunk 5]
```diff
@@ -77,6 +83,56 @@ export default function setupEventListeners(store, canvas) {
...
+          if (state.ui.selection.selectedGoals.length) {
+            // use first
+            store.dispatch(setGoalClone(state.ui.selection.selectedGoals))
+          }
+        }
+        break
+      case 'KeyV':
+        if (state.ui.keyboard.ctrlKeyDown) {
+          if (state.ui.goalClone.goals.length) {
+            state.ui.goalClone.goals.forEach(value => {
```
Comment (Connoropolous) [inline src/event-listeners/index.js]: I think this part should be refactored... 
create a new file somewhere, that exports a function that accepts the necessary arguments and performs these actions... 
e.g. 
`cloneGoals.js`
```javascript

export default function cloneGoals(store, goalsToClone, goalMembers) {
  ...
}
```

**Review by Connoropolous — CHANGES_REQUESTED** [review]
On a great track Hector. 
I have an additional questions for @pegahvaezi 
Copy-Paste is currently cloning 
1. the goals themselves
2. the members/squirrels of the selected goals
Should it also copy the
1. the priority votes?
2. the edges between Goals?

File: src/components/cloneGoals.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import { createGoal } from '../goals/actions'
+import { selectGoal } from '../selection/actions'
+import { addMemberOfGoal } from '../goal-members/actions'
+import moment from 'moment'
+
+export default function cloneGoals(store, goalsToClone, goalMembers, goals) {
```
Comment (Connoropolous) [inline src/components/cloneGoals.js]: Does `goals` get used anywhere? remove if not

File: src/event-listeners/index.js [hunk 7]
```diff
@@ -29,11 +31,15 @@ import {
...
-import { archiveGoal } from '../goals/actions'
+import { archiveGoal, createGoal } from '../goals/actions'
 import { setScreenDimensions } from '../screensize/actions'
 import { changeTranslate, changeScale } from '../viewport/actions'
 
 import layoutFormula from '../drawing/layoutFormula'
+import { setGoalClone } from '../goal_clone/actions'
+import moment from 'moment'
+import { addMemberOfGoal } from '../goal-members/actions'
+import cloneGoals from '../components/cloneGoals'
```
Comment (Connoropolous) [inline src/event-listeners/index.js]: Can you move `cloneGoals.js` file into the same folder as this file I guess. `event-listeners`

**Review by Connoropolous — CHANGES_REQUESTED** [review]
Right idea! Just put it in the same folder I guess

File: src/components/cloneGoals.js (hunk 6 above)
Comment (hector0513) [reply src/components/cloneGoals.js]: `.dispatch(
        createGoal.create({
          goal: {
            ...goals[value],
            timestamp_created: moment().unix(),
          },
          maybe_parent_address: null,
        })
      )`

Comment (Connoropolous) [reply src/components/cloneGoals.js]: oh right, I see 👍

File: src/event-listeners/index.js (hunk 5 above)
Comment (pegahvaezi) [reply src/event-listeners/index.js]: re: questions 3 and 4 : no cloning of priority votes (yet) and no copying of edges. If using command + D, the new card should appear the closest possible to the original card.

File: src/reducer.js [hunk 8]
```diff
@@ -24,6 +24,7 @@ import viewport from './viewport/reducer'
 import expandedView from './expanded-view/reducer'
 import goalVotes from './goal-votes/reducer'
 import goalComments from './goal-comments/reducer'
+import goalClone from './goal_clone/reducer'
```
Comment (Connoropolous) [inline src/reducer.js]: I know this seems trivial, but can you make the `goal_clone` folder `goal-clone`
Sorry!

**Review by Connoropolous — APPROVED** [review]
I'll change this to approve, but still, if you can make the folder name change before you merge that would be good

**Comment by Connoropolous** [comment]
Pegah says that we don't need to worry about cloning PriorityVotes or Edges yet, so this is fine 👍

File: src/event-listeners/index.js [hunk 9]
```diff
@@ -34,6 +36,9 @@ import { setScreenDimensions } from '../screensize/actions'
 import { changeTranslate, changeScale } from '../viewport/actions'
 
 import layoutFormula from '../drawing/layoutFormula'
+import { setGoalClone } from '../goal_clone/actions'
```
Comment (Connoropolous) [inline src/event-listeners/index.js:39]:
```suggestion
import { setGoalClone } from '../goal-clone/actions'
```

**Pull request merged**